pkg/skaffold/build/gcb/cloud_build.go

```diff
@@ -115,30 +136,46 @@ func (b *Builder) buildArtifactWithCloudBuild(ctx context.Context, out io.Writer
 	if artifact.JibArtifact != nil {
 		deps, err := jibAddWorkspaceToDependencies(artifact.Workspace, dependencies)
 		if err != nil {
-			return "", fmt.Errorf("walking workspace for Jib projects: %w", err)
+			return "", sErrors.NewErrorWithStatusCode(&proto.ActionableErr{
+				ErrCode: proto.StatusCode_BUILD_GCB_JIB_DEPENDENCY_ERR,
+				Message: fmt.Sprintf("walking workspace for Jib projects: %s", err),
+			})
 		}
 		dependencies = deps
 	}
 
 	if err := sources.UploadToGCS(ctx, c, artifact, cbBucket, buildObject, dependencies); err != nil {
-		return "", fmt.Errorf("uploading source archive: %w", err)
+		return "", sErrors.NewErrorWithStatusCode(&proto.ActionableErr{
+			ErrCode: proto.StatusCode_BUILD_GCB_UPLOAD_TO_GCS_ERR,
+			Message: fmt.Sprintf("uploading source archive: %s", err),
+		})
 	}
 
 	buildSpec, err := b.buildSpec(ctx, artifact, tag, cbBucket, buildObject)
 	if err != nil {
-		return "", fmt.Errorf("could not create build description: %w", err)
+		return "", sErrors.NewErrorWithStatusCode(&proto.ActionableErr{
+			ErrCode: proto.StatusCode_BUILD_GCB_GET_BUILD_DESCRIPTOR_ERR,
```

Comment:
I wonder if this might be better called GENERATE rather than GET (but avoiding CREATE due to BUILD_GCB_CREATE_BUILD_ERR below)